docs/resources/aws_security_group.md.erb

```diff
@@ -224,55 +234,78 @@ The `allow` series of matchers enable you to perform queries about what network
+
+  - position - A one-based index into the list of rules. If provided, this restricts the evaluation to the rule at that position. You may also use the special values `:first` and `:last`. `position` may also be used to enable `allow_in_only` and `allow_out_only` to work with multi-rule Security Groups.
+
+  - protocol - Specifies the IP protocol. 'tcp', 'udp', and 'icmp' are some typical values. The string "-1" or 'any' is used to indicate any protocol.
+
+  - to\_port - Determines if a rule exists whose port range ends at the specified number. The word 'to\_' does *not* relate to inbound/outbound directionality; it relates to the port range ("counting *to*"). `to_port` is an exact criterion; so if the rule allows 1000-2000 and you specify a `to_port` of 1999, it does not match.
+    
+    describe aws\_security\_group(group\_name: 'mixed-functionality-group') do
+    
+    # Allow RDP from defined range
+    
+    it { should allow\_in(port: 3389, ipv4\_range: '10.5.0.0/16') }
+    
+    # Allow SSH from two ranges
+    
+    it { should allow\_in(port: 22, ipv4\_range: \['10.5.0.0/16', '10.2.3.0/24'\]) }
+    
+    # Check Bacula port range
+    
+    it { should allow\_in(from\_port: 9101, to\_port: 9103, ipv4\_range: '10.6.7.0/24') }
+    
+    # Assuming the AWS SG allows 9001-9003, use inexact matching to check 9002
+    
+    it { should allow\_in(port: 9002) }
+    
+    # Assuming the AWS SG allows 10.2.1.0/24, use inexact matching to check 10.2.1.33/32
+    
+    it { should allow\_in(ipv4\_range: '10.2.1.33/32') }
+    
+    # Ensure the 3rd outbound rule is TCP-based
+    
+    it { should allow\_in(protocol: 'tcp', position: 3') }
+    
+    # Do not allow unrestricted IPv4 access.
+    
+    it { should\_not allow\_in(ipv4\_range: '0.0.0.0/0') }
     end
-
+    
     # Suppose you have a Group that should allow SSH and RDP from
+    
     # the admin network, 10.5.0.0/16. The resource has 2 rules to
+    
     # allow this, and you want to ensure no others have been added.
-    describe aws_security_group(group_name: 'admin-group') do
-      # Allow RDP from a defined range and nothing else
-      # The SG must have this rule in position 1 and it must match this exactly
-      it { should allow_in_only(port: 3389, ipv4_range: '10.5.0.0/16', position: 1) }
-
-      # Specify position 2 for the SSH rule.  Without `position`,
-      # allow_in_only only allows one rule, total.
-      it { should allow_in_only(port: 22, ipv4_range: '10.5.0.0/16', position: 2) }
-
-      # Because this is an _only matcher, this fails - _only matchers
-      # use exact IP matching.
-      it { should allow_in_only(port: 3389, ipv4_range: '10.5.1.34/32', position: 1) }
+    
+    describe aws\_security\_group(group\_name: 'admin-group') do
```

Comment:
I think the lines below need to be cleaned up by using backticks and not escaping the underscores.